src/app/admin/websiteDataTable/WebsiteDataTable.tsx

```diff
@@ -0,0 +1,231 @@
+"use client";
+
+import React, { useEffect, useState } from "react";
+import supabase from "@/supabaseClient";
+import {
+    DataGrid,
+    GridActionsCellItem,
+    GridColDef,
+    GridEventListener,
+    GridRowEditStopReasons,
+    GridRowId,
+    GridRowModes,
+    GridRowModesModel,
+    useGridApiRef,
+} from "@mui/x-data-grid";
+import EditIcon from "@mui/icons-material/Edit";
+import SaveIcon from "@mui/icons-material/Save";
+import CancelIcon from "@mui/icons-material/Close";
+import { LinearProgress } from "@mui/material";
+import { DatabaseError } from "@/app/errorClasses";
+import { fetchWebsiteData, updateDbWebsiteData } from "./FetchWebsiteData";
+import CustomComponent from "./CustomEditComponent";
+import { logErrorReturnLogId, logInfoReturnLogId } from "@/logger/logger";
+
+export interface WebsiteDataRow {
+    readableName: string;
+    id: string;
+    value: string;
+}
+
+const WebsiteDataTable: React.FC = () => {
+    const [rows, setRows] = useState<WebsiteDataRow[]>([]);
+    const [rowModesModel, setRowModesModel] = useState<GridRowModesModel>({});
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+    const apiRef = useGridApiRef();
```

Comment:
Could you explain why we need this?

Reply:
We use it as a context which allows us to update the value of the edited cell within the custom component by using apiRef.current.setEditCellValue(....)
I think this is the conventional way to deal with custom edit components in the mui data grid
https://mui.com/x/react-data-grid/editing/#create-your-own-edit-component

Reply:
makes sense!